entrypoint.sh

```diff
@@ -1,46 +1,75 @@
-#!/bin/sh -l
+#!/bin/bash
+set -e
 
-sourceDirectory=$1
-platform=$2
-platformVersion=$3
+checkArg() {
+    # Check if the input has content and is not just whitepsace or is empty
+    case $1 in
+    (*[![:blank:]]*) echo "$1";;
+    (*) echo ""
+    esac
+}
 
-echo
+getAbsolutePath(){
+    local path="$1"
+
+    if [ -z "$path" ]; then
+        echo "$PWD"
+    fi
+
+    # Check if the path is relative or absolute.
+    # If relative path is given, create an absolute path
+    case $path in
+    /*) echo $path ;;
+    *) echo "$PWD/$path" ;;
+    esac
+}
 
-if [ -n "${sourceDirectory}" ]
+sourceDirectory=$(checkArg "$1")
+platform=$(checkArg "$2")
+platformVersion=$(checkArg "$3")
+outputDirectory=$(checkArg "$4")
+
+echo
+if [ -z "$sourceDirectory" ]
 then
-    sourceDirectory="$PWD/$sourceDirectory"
-    echo "Relative path to source directory provided -- the following directory will be built: '${sourceDirectory}'"
+    sourceDirectory="$PWD"
+    echo "No source directory was provided -- the root of the repository ('GITHUB_WORKSPACE' environment variable) will be built: '${sourceDirectory}'"
 else
-    sourceDirectory=$PWD
-    echo "No source directory provided -- the root of the repository ('GITHUB_WORKSPACE' environment variable) will be built: '${sourceDirectory}'"
+    sourceDirectory=$(getAbsolutePath "$sourceDirectory")
+    echo "Source directory was provided. Following directory will be built: '${sourceDirectory}'"
 fi
 
 echo
-oryxCommand="oryx build ${sourceDirectory}"
+oryxCommand="oryx build $sourceDirectory"
 
-echo
-
-if [ -n "${platform}" ]
+if [ -z "$outputDirectory" ]
 then
-    echo "Platform provided: '${platform}'"
-    oryxCommand="${oryxCommand} --platform ${platform}"
+    echo "No output directory was provided"
 else
-    echo "No platform provided -- Oryx will enumerate the source directory to determine the platform."
+    outputDirectory=$(getAbsolutePath "$outputDirectory")
+    oryxCommand="$oryxCommand -o $outputDirectory"
```

Comment:
nit: just declare the oryxcommand as a variable may be?

Reply:
Can you clarify?...`oryxCommand` is already a variable